modules/querying/pages/query-optimizer/index.adoc

```diff
@@ -0,0 +1,24 @@
+= Query Optimizer
+:description: Overview of the query optimizer.
+
+NOTE: This is a Preview feature and should not be used for production purposes.
+
+The query optimizer optimizes the execution plan of a query based on cost.
+
+When you write a GSQL query with a xref:querying:select-statement/index.adoc#_path_pattern[path pattern (Syntax V2)], you specify a pattern that contains a source vertex set, one or more path edge patterns, and one or more step vertex sets.
+This pattern is declarative.
+In other words, it specifies the structure of paths to be matched, without specifying the particulars of how the graph is traversed to find these paths.
+
+When you xref:query-operations.adoc#_install_query[install the GSQL query], GSQL compiles the query logic into a binary execution plan.
+The query optimizer takes statistics such as the number of instances for each vertex and edge type, as well as the distribution of values in the attributes relevant to the query.
+It then optimizes the execution plan to traverse the graph based on those statistics to minimize the cost of executing the query.
```

Comment:
I think we should merge this paragraph with the previous one. I don't think we need the first sentence. I think it's prior knowledge that we can assume the user already has if they're reading this, since it's called out in other areas of the docs.

I also don't think we need to be explicit about the various statistics here. You've been explicit about those elsewhere, so I think it's fine to say
```
It is the query optimizer's job to determine the best graph traversal algorithm to satisfy the query.
It does so by utilizing various pre-computed statistics describing the data distribution of the types referenced in the query.
It then optimizes the execution plan to traverse the graph based on those statistics to minimize the cost of executing the query.
```

Reply:
We cannot assume everyone reading this page has this knowledge. The best practice for information architecture for online documentation which we (try to) follow is called Every Page is Page One. People can stumble on the page from many sources for any reason - we should always try to provide references to background information they need to understand the page.